### DISCUSSION TITLE
Emails should have an individual "private" option

### DISCUSSION THREAD

**Discussion by loricvdt** [community/community#58127] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Since multiple emails can be used in Github, there should be a way to select which one we want to keep private.

For an example, I wish to keep my personal primary email private but still commit with my work email with my work-related projects.
There is no way to do that right now, it is either all emails are private, or all are considered public.

**Comment by Gustavonn07** [comment]
One way that I see how to fix this by hand would be to deprive the emails and in your github readme putting the email you want to use for contact

Hope it helps!

**Comment by hafiz-muhammad** [comment]
You can create an email alias using sites like [SimpleLogin](https://simplelogin.io/) to protect your email address. All emails sent to the alias address will be forwarded to your actual email address.

**Comment by preston-work** [comment]
I wish 'account' and 'commit' email addresses operated independently. That way only 'commit' email addresses get exposed publicly (in the repo). Non-commit email addresses could get block from pushes, and an optional check box next to each 'account' email could allow it to operate as a 'commit' email but otherwise those stay private. In addition, specifying per org 'commit' email addresses would allow for even more flexibility, and avoid the possibility of accidentally using the wrong email address.

The way 'account' and 'commit' emails work together now makes them very intertwined. The design could use some improvements. It seems not the take into account the different ways people use their email addresses. Some people prefer not making their primary email address 'public', but would like to use a secondary address for commits (and not just the Github no-reply address). These secondary emails could link to a work account or just an additional email for commits, while still retaining control of the account via a personal (and private) primary email, as well as, protecting that personal email from getting accidentally used as a 'commit' email.

**Comment by sergejs-katusenoks** [comment]
As an Enterprise user I think it would be a good feature. We have a pattern check that you have to make pushes from our domain only. It works fine until you try to merge with the "Keep my email addresses private" enabled. 

I believe this setting should be moved to each email. This will insure privacy for any personal email and allow visibility for Enterprise/Org users. Make sense to label this as a feature request.
